### PR TITLE
Set up Mailers and Notifications for Models/Controllers Pt.1

### PR DESCRIPTION
**Pull request by hvillero**
Notifications to add:
 - MODEL - User for Create and Update
Mailers to add:
- UserMailer
- Add seeds
- Update Readme

### CODE REVIEW

File: app/notifications/user_sing_up_notification.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class UserSingUpNotification < Noticed::Base
```
Comment (pJeyakumar) [inline app/notifications/user_sing_up_notification.rb:1]:
```suggestion
class UserSignUpNotification < Noticed::Base
```

Spelling error in class

File: app/notifications/user_sing_up_notification.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  # Variables that tells the system if it should send a notification if the recipient isn't subscribed
+  DEFAULT_SEND_NOTIFICATION = true
+  DEFAULT_SEND_EMAIL_NOTIFICATION = true
+
+  # Add required params
+  param :event_user
+
+  def message
+    "A new user #{params[:event_user]} with email #{params[:event_user].email} has been Sign Up."
```
Comment (pJeyakumar) [inline app/notifications/user_sing_up_notification.rb:17]:
```suggestion
    "A new user #{params[:event_user]} with email #{params[:event_user].email} has signed up."
```
Minor grammar correction

File: app/notifications/user_update_notification.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  # Variables that tells the system if it should send a notification if the recipient isn't subscribed
+  DEFAULT_SEND_NOTIFICATION = true
+  DEFAULT_SEND_EMAIL_NOTIFICATION = true
+
+  # Add required params
+  param :event_user
+
+  def message
+    "The user #{params[:event_user].to_s} - #{params[:event_user].email} has been updated."
```
Comment (pJeyakumar) [inline app/notifications/user_update_notification.rb:17]:
```suggestion
    "The user #{params[:event_user]} - #{params[:event_user].email} has been updated."
```
No need to call `to_s` explicitly

File: app/views/aircraft_mailer/created_email.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @user.full_name %>,</p>
```
Comment (pJeyakumar) [inline app/views/aircraft_mailer/created_email.html.erb:1]:
```suggestion
<p>Hello <%= @user %>,</p>
```
`User` does not have a `full_name` attribute or method, can just call `@user`

File: app/views/user_mailer/created_user.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @user %>,</p>
+<br />
+
+<p> The user <%= @event_user.to_s %> has been created.</p>
```
Comment (pJeyakumar) [inline app/views/user_mailer/created_user.html.erb:4]:
```suggestion
<p> The user <%= @event_user%> has been created.</p>
```

Do not need to explicitly call `to_s` here

File: app/views/user_mailer/update_user.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @user %>,</p>
+<br />
+
+<p> The user <%= @event_user.to_s %> has been updated.</p>
```
Comment (pJeyakumar) [inline app/views/user_mailer/update_user.html.erb:4]:
```suggestion
<p> The user <%= @event_user %> has been updated.</p>
```

Do not need to explicitly call `to_s` here.

File: app/views/user_mailer/welcome_email.html.erb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />
+</head>
+
+<body>
+    <h1>Welcome, <%= @user.full_name %>
```
Comment (pJeyakumar) [inline app/views/user_mailer/welcome_email.html.erb:9]:
```suggestion
    <h1>Welcome, <%= @user %>
```

`User` does not have a `full_name` method

File: app/mailers/user_mailer.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
+class UserMailer < ApplicationMailer
```
Comment (pJeyakumar) [inline app/mailers/user_mailer.rb:1]: I believe there is an issue with the User emailing, after creating a new user, I do not see a Sign Up Email notification in `/mail` (perhaps there are errors in the `created_user.html.erb`)

**Review by pJeyakumar — COMMENTED** [review]
There are a few minor changes to be made.
I believe the User Email notifications are not working correctly, when going to `/mail` after creating a new user, there is no "Sign Up Email" present.

File: app/mailers/aircraft_mailer.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
+class AircraftMailer < ApplicationMailer
+  before_action :set_user
+  before_action :set_aircraft
+
+
+  def created_email
+    mail(to: @user.email, subject: "[DIGITAL FLIGHT SIM] New Aircraft Created: #{@aircraft.name}")
```
[the remark this answers is not in the document]
Comment (hvillero) [reply app/mailers/aircraft_mailer.rb:7]: good catch , thx

File: app/mailers/user_mailer.rb (hunk 8 above)
Comment (hvillero) [reply app/mailers/user_mailer.rb:1]: mmm, maybe the let me check
```ruby
  users_url(@event_user)
```

**Review by srb- — APPROVED** [review]
:+1:

File: app/notifications/user_sing_up_notification.rb
Comment (pJeyakumar) [inline app/notifications/user_sing_up_notification.rb:1]: I believe you'll also need to change the name of this file to match the class

File: app/mailers/user_mailer.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def update_user    
+    mail(to: @user.email, subject: "[DIGITAL FLIGHT SIM] New Aircraft Created: #{@event_user}")
+  end
+
+  private
+
+  def set_user
+    @user = params[:user]
```
Comment (pJeyakumar) [inline app/mailers/user_mailer.rb:21]:
```suggestion
    @user = params[:recipient]
```

parameter for mailer should be `recipient`

**Review by pJeyakumar — COMMENTED** [review]
We're almost there! Just 2 more quick fixes and it should be working soon!

**Review by pJeyakumar — APPROVED** [review]

**Pull request merged**